client/keys.go

```diff
@@ -688,7 +688,11 @@ func (k *Key) unsealDirect(in *pb.SealedBytes, opts unsealOptsDirect) ([]byte, e
 		return nil, fmt.Errorf("failed to create PCRSelection: %v", err)
 	}
 
-	sess, cleanup2, err := tpm2direct.PolicySession(k.transportTPM(), sessionHashAlgTpmDirect, SessionHashAlg.Size())
+	var options []tpm2direct.AuthOption
+	options = append(options, tpm2direct.Salted(tpm.Handle(k.handle), k.pubAreaDirect))
```

Comment:
should we check the key is a decrypt key? we should probably double check it's asymmetric too, otherwise the user needs to create a sym key with a sensitive part that can be sniffed on the bus

Reply:
I also wonder if we should check the type of the key is RSA or ECC? otherwise we get some error of the form "failed to start Policy Session: salt encryption alg '%v' not supported", which is not very descriptive. the problem is we duplicate the supported types in tools as well as direct, but I'm inclined to give better error messages a higher priority.

Reply:
Resolved the decrypt check.